### PR TITLE
Arc - support @All List<X> synthetic injection points

### LINKED ISSUE #42834: ArC: synthetic injection points should support `@Inject @All List<>`
### Description

- https://quarkus.io/guides/cdi-integration#synthetic-injection-points
- https://quarkus.io/guides/cdi-reference#injecting-multiple-bean-instances-intuitively

Zulip discussion: https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/.E2.9C.94.20Synthetic.20vs.20CDI.20Beans.20in.20Extension

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by manovotn**
Fixes #42834

### CODE REVIEW

File: independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/buildextension/beans/SyntheticInjectionPointListAllTest.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+                    .done();
+        }
+
+    }
+
+    public static class SynthBeanCreator implements BeanCreator<SyntheticBean> {
+
+        @Override
+        public SyntheticBean create(SyntheticCreationalContext<SyntheticBean> context) {
+            return new SyntheticBean(context.getInjectedReference(new TypeLiteral<>() {
```
Comment (mkouba) [inline independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/buildextension/beans/SyntheticInjectionPointListAllTest.java:83]: Hm, this is translated to the required type `Object`, or? So I wonder how the test could pass :thinking:.

File: independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/buildextension/beans/SyntheticInjectionPointListAllTest.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            .beanClasses(SomeBean.class)
+            .beanRegistrars(new TestRegistrar()).build();
+
+    @Test
+    public void testListAllInjection() {
+        SyntheticBean synthBean = Arc.container().instance(SyntheticBean.class).get();
+        assertNotNull(synthBean);
+        List<SomeBean> list = synthBean.getList();
+        assertNotNull(list);
+        assertTrue(list.size() == 1);
```
Comment (mkouba) [inline independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/buildextension/beans/SyntheticInjectionPointListAllTest.java:38]: `assertEquals(1, list.size())` is usually a bit better - at least you can see the actual size in the log ;-)

File: independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/buildextension/beans/SyntheticInjectionPointListAllTest.java (hunk 1 above)
Comment (Ladicek) [reply independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/buildextension/beans/SyntheticInjectionPointListAllTest.java:83]: The `SyntheticBean` constructor takes a `List<SomeBean>`, so the type argument here is inferred to be `List<SomeBean>`.

Comment (mkouba) [reply independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/buildextension/beans/SyntheticInjectionPointListAllTest.java:83]: Ah! Not sure if inferring makes the code more readable here though.

Comment (manovotn) [reply independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/buildextension/beans/SyntheticInjectionPointListAllTest.java:83]: > Ah! Not sure if inferring makes the code more readable here though.

I can see why it is confusing - I had the type here explicitly but IDEA told me it isn't needed.
Let me add it back :)

**Comment by manovotn** [comment]
Updated PR as per comments

**Review by Ladicek — APPROVED** [review]

**Review by mkouba — APPROVED** [review]

**Pull request merged**